presto-record-decoder/src/main/java/com/facebook/presto/decoder/thrift/ThriftFieldDecoder.java

```diff
@@ -0,0 +1,369 @@
+        public final boolean accept(DecoderColumnHandle columnHandle)
+        {
+            return this.columnHandle.equals(columnHandle);
+        }
+
+        @Override
+        public final boolean isNull()
+        {
+            return value == null;
+        }
+
+        @Override
+        public boolean getBoolean()
+        {
+            return isNull() ? false : (Boolean) value;
+        }
+
+        @Override
+        public long getLong()
+        {
+            return isNull() ? 0L : getLongExpressedValue(value);
+        }
+
+        private static long getLongExpressedValue(Object value)
+        {
+            if (value instanceof Date) {
+                long storageTime = ((Date) value).getTime();
+                // convert date from VM current time zone to UTC
+                long utcMillis = storageTime + DateTimeZone.getDefault().getOffset(storageTime);
+                return TimeUnit.MILLISECONDS.toDays(utcMillis);
+            }
+            if (value instanceof Timestamp) {
+                long parsedJvmMillis = ((Timestamp) value).getTime();
+                DateTimeZone jvmTimeZone = DateTimeZone.getDefault();
+                long convertedMillis = jvmTimeZone.convertUTCToLocal(parsedJvmMillis);
+
+                return convertedMillis;
+            }
+            if (value instanceof Float) {
+                return floatToRawIntBits(((Float) value));
+            }
+            return ((Number) value).longValue();
+        }
+
+        @Override
+        public double getDouble()
+        {
+            return isNull() ? 0.0d : (Double) value;
+        }
+
+        @Override
+        public Slice getSlice()
+        {
+            return isNull() ? EMPTY_SLICE : getSliceExpressedValue(value, columnHandle.getType());
+        }
+
+        private static Slice getSliceExpressedValue(Object value, Type type)
+        {
+            Slice sliceValue;
+            if (value instanceof String) {
+                sliceValue = Slices.utf8Slice((String) value);
+            }
+            else if (value instanceof byte[]) {
+                sliceValue = Slices.wrappedBuffer((byte[]) value);
+            }
+            else if (value instanceof Integer) {
+                sliceValue = Slices.utf8Slice(value.toString());
+            }
+            else {
+                throw new IllegalStateException("unsupported string field type: " + value.getClass().getName());
+            }
+            if (isVarcharType(type)) {
+                sliceValue = truncateToLength(sliceValue, type);
+            }
+            if (isCharType(type)) {
+                sliceValue = trimSpacesAndTruncateToLength(sliceValue, type);
+            }
+
+            return sliceValue;
+        }
+
+        @Override
+        public Block getBlock()
+        {
+            if (isNull()) {
+                return null;
+            }
+
+            Type type = columnHandle.getType();
+            return serializeObject(type, null, value);
```

Comment:
serialized object could be null, but in that case it will just show null in the query result, not causing anything to crash.